samples/dnn/speech_recognition.py

```diff
@@ -0,0 +1,455 @@
+    backends = (cv.dnn.DNN_BACKEND_DEFAULT, cv.dnn.DNN_BACKEND_HALIDE, cv.dnn.DNN_BACKEND_INFERENCE_ENGINE, cv.dnn.DNN_BACKEND_OPENCV)
+    # Target Devices for computation
+    targets = (cv.dnn.DNN_TARGET_CPU, cv.dnn.DNN_TARGET_OPENCL, cv.dnn.DNN_TARGET_OPENCL_FP16)
+
+    parser = argparse.ArgumentParser(description='This script runs Jasper Speech recognition model',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--input_audio', type=str, help='Path to input audio file.')
+    parser.add_argument('--show_spectrogram', action='store_true', help='Whether to show a spectrogram of the input audio.')
+    parser.add_argument('--model', type=str, default='jasper.onnx', help='Path to the onnx file of Jasper. default="jasper.onnx"')
+    parser.add_argument('--output', type=str, help='Path to file where recognized audio transcript must be saved. Leave this to print on console.')
+    parser.add_argument('--backend', choices=backends, default=cv.dnn.DNN_BACKEND_DEFAULT, type=int,
+                        help='Select a computation backend: '
+                        "%d: automatically (by default) "
+                        "%d: Halide"
+                        "%d: OpenVINO Inference Engine "
+                        "%d: OpenCV Implementation " % backends)
+    parser.add_argument('--target', choices=targets, default=cv.dnn.DNN_TARGET_CPU, type=int,
+                        help='Select a target device: '
+                        "%d: CPU target (by default) "
+                        "%d: OpenCL "
+                        "%d: OpenCL FP16 " % targets)
+
+    args, _ = parser.parse_known_args()
+
+    if args.input_audio and not os.path.isfile(args.input_audio):
+        raise OSError("Input audio file does not exist")
+    if not os.path.isfile(args.model):
+        raise OSError("Jasper model file does not exist")
+
+    # Read audio File
+    try:
+        audio = sf.read(args.input_audio)
+        X=audio[0]
+        seq_len=np.array([X.shape[0]],dtype=np.int32)
+    except:
+        raise Exception(f"Soundfile cannot read {args.input_audio}. Try a different format")
+
+    # Load Network
+    net = cv.dnn.readNetFromONNX(args.model)
+    net.setPreferableBackend(args.backend)
+    net.setPreferableTarget(args.target)
+
+    # Get Filterbank Features
+    feature_extractor = FilterbankFeatures()
+    features = feature_extractor.calculate_features(x=X,seq_len=seq_len)
+
+    # Show spectogram if required
+    if args.show_spectrogram:
+        img = cv.normalize(src=features[0], dst=None, alpha=0, beta=255, norm_type=cv.NORM_MINMAX, dtype=cv.CV_8U)
+        img = cv.applyColorMap(img,cv.COLORMAP_JET)
+        cv.imshow('spectogram',img)
+        cv.waitKey(0)
+
+    # Initialize decoder
+    decoder = Decoder()
+
+    # make prediction
+    net.setInput(features)
+    output = net.forward()
+
+    # decode output to transcript
+    prediction = decoder.decode(output[0])
```

Comment:
Why output[0] is needed?
Is the first dimension always equal to 1? For any batch size?

Reply:
Since this script handles a single audio file, which is treated as batch size 1, the input to jasper is always of shape (1,64,?) and gives an output of shape (1,?,29).

Reply:
Could you please check how model works with several audio files (batch size >1)?